frontend/src/components/session-in-progress/session-progress/session-progress.tsx

```diff
@@ -15,6 +21,19 @@ const SessionProgress = (props: Props): ReactElement => {
   const questionProgress = state.sessionQuestions[questionNumber].progress;
   const isClosed = state.sessionQuestions[questionNumber].isClosed;
 
+  // do this to compare objects
+  // because useEffect doesn't do deep compare
+  const effectDependency = JSON.stringify(state.sessionQuestions);
+
+  useEffect(() => {
+    if (
+      state.sessionQuestions.every((question: Question) => question.isClosed)
```

Comment:
I notice you're using `isClosed` of each sessionQuestion to determine if all questions have been responded to. I like this idea, but it's not always working. I think it's because of the incorrect dispatch i mentioned above ^